Speed_Controller/src/main.c

```diff
@@ -0,0 +1,167 @@
+#include "stm32f10x.h"
+
+#include "ADC/ADC.h"
+#include "Encoder/encoder.h"
+#include "CAN.h"
+#include "virtual_com.h"
+#include "Timer/Timer.h"
+
+#define DRIVE_CONTROL_ID 0x500
+
+union {
+	float float_var;
+	uint8_t chars[4];
+} u;
+
+union {
+	float float_var;
+	uint8_t chars[4];
+} v;
+
+int main(void){
+	CANInit();
+	EncoderInit();
+	ADCInit();
+	TimerInit(100);
```

Comment:
we should probably try increasing the period until the motor controller stops operating smoothly